Sources/XMLCoder/Decoder/XMLKeyedDecodingContainer.swift

```diff
@@ -192,6 +174,31 @@ struct XMLKeyedDecodingContainer<K: CodingKey>: KeyedDecodingContainerProtocol {
     }
 }
 
+extension XMLKeyedDecodingContainer {
+    internal init?(box: Box, decoder: XMLDecoderImplementation) {
+        if let keyedContainer = box as? KeyedContainer {
```

Comment:
Should be a `switch` instead. Switch statements are a bit faster.

Reply:
```swift
switch box {
case let container as KeyedContainer:
...
default:
    return nil
}
```

Reply:
done